pkg/doc/verifiable/common.go

```diff
@@ -55,16 +55,44 @@ func (ja JWSAlgorithm) jose() (jose.SignatureAlgorithm, error) {
 // Proof defines embedded proof of Verifiable Credential
 type Proof interface{}
 
-type typedID struct {
+// TypedIDExtraFields is a map of extra fields which extend typed ID field.
+type TypedIDExtraFields map[string]interface{}
+
+// TypedID defines a flexible structure with id and name fields and arbitrary extra fields
+// kept in TypedIDExtraFields.
+type TypedID struct {
 	ID   string `json:"id,omitempty"`
 	Type string `json:"type,omitempty"`
+
+	TypedIDExtraFields `json:"-"`
```

Comment:
Repetition of names here: TypedID.TypedIDExtraFields

Can this just be TypedID.ExtraFields?

Reply:
Sure, this will sound better, no need for type embedding. I will change
```
TypedIDExtraFields `json:"-"`
```
to 
```
ExtraFields TypedIDExtraFields `json:"-"`
```

Reply:
Looks good! (I assume the extra "TypedIDExtraFields" in the second line is a typo)

Reply:
Oh I see, you wanted to extra TypedIDExtraFields into a type

Reply:
Ah, right. Extra fields are defined for two types - `rawCredential` and `TypedID`. Why not have single `ExtraField` type declaration used by these two types? I will declare it as 
```
type ExtraFields map[string]interface{}
```

Reply:
Gotcha, makes sense. You can disregard my other question I made asking about this since you answered it here